text/0000-deprecate-globals-resolver.md

```diff
@@ -0,0 +1,115 @@
+Using it looks like the following:
+
+```js
+// app.js
+var App = Ember.Application.create();
+
+App.Router.map(function() {
+  this.route('about');
+});
+
+App.AboutRoute = Ember.Route.extend({
+  model: function() {
+    return ['red', 'yellow', 'blue'];
+  }
+});
+```
+
+```html
+// index.html
+<script type="text/x-handlebars" data-template-name="about">
+  <ul>
+    {{#each model as |item|}}
+      <li>{{item}}</li>
+    {{/each}}
+  </ul>
+</script>
+```
+
+# Transition Path
+
+Primarily, the transition path is to recommend using Ember-CLI.
+
+During the 3.x timeframe, it MAY become increasingly difficult to use this old functionality.
+For example, with the release of 3.0, we already stopped publishing builds that support
+globals mode. Here are some of the changes that have impacted or may soon impact users of globals mode:
+
+## Impact of ES6 modules
+
+Users of ES6 modules must use their own build tooling to convert them to named AMD modules via Babel.
+No support is provided for &lt;script type="module"&gt; at this time, although that may change.
+
+## Impact of New Module Imports
+
+Globals based apps are only able to use new module imports via the polyfill available at
+https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill No build support for this is provided.
+
+## Impact of not publishing globals builds
+
+It is necessary to get a globals build of Ember.js from the npm package now that globals builds
+are no longer published to S3, builds.emberjs.com, and CDNs.
+
+## Impact of not Generating a Globals Build in Ember.js Package
+
+At some point during the 3.x cycle, it may be that we no longer publish a globals build in the
+npm package. At that point, it may become necessary to use Ember-CLI to generate a globals build
+of Ember.js
+
+## Impact of Package Splitting
+
+Work has started on package splitting. It is likely that the globals resolver may not be included
+in a default partial build of Ember.js and may be moved to its own package for easy removal.
+
+## Impact of Tree Shaking
+
+If the globals resolver is moved to a separate package, it will likely not be included in a build
+of Ember.js by default unless tree shaking is turned off.
+
+# How We Teach This
+
+We already do teach this and don't teach the globals resolver. No changes required here.
+
+# Drawbacks
+
+A drawback is that people may want alternate build tooling to Ember-CLI.
+We have mitigated this by openly publishing the ember-cli resolver and all parts of the
+ember-cli ecosystem under the MIT license.
+Alternate build tooling may simply use this open source code to build a competing
+infrastructure to ember-cli.
+
+# Alternatives
+
+Without doing this, we will have to continue to ship and maintain this rarely used functionality.
+We don't believe this is a reasonable alternative.
+
+# Unresolved questions
+
+There has never been a transition guide for transitioning an old codebase to Ember-CLI.
+Do we want to create one at this late date?
```

Comment:
I think that we certainly _could_ do it, but I don't think that it would be a requirement for moving forward with this RFC.  As it stands today, the migration path away from globals mode to a more natural ember-cli built application is something _most_ of our community has already absorbed, and the effort of writing a thorough migration guide at this point far outweighs the upside...

Reply:
I think a traditional approach to deprecation RFCs is that we write the deprecation documentation here in the RFC. Once we pick a deprecation strategy I'd like to see that added.

Reply:
I would appreciate it if anyone can PR some documentation against this RFC.

I'm really not sure what to say except "Do not use globals resolver, use ember-cli".

Reply:
@Gaurav0 put together a summary for the deprecation guide over in https://github.com/ember-learn/deprecation-app/issues/153